extensions/wikia/EditPageLayout/js/editor/AceEditor.js

```diff
@@ -86,18 +86,39 @@ define(
 	 * Init submit action
 	 */
 	function initSubmit() {
-		$('#editform').submit(function(e) {
-			var hiddenInput = ace.getInput().val(ace.getContent());
+		$('#editform').submit(function (e) {
+			var hiddenInput = ace.getInput().val(ace.getContent()),
+				$editForm = $(this);
 
 			disableBeforeUnload = true;
 
-			$(this).unbind('submit').append(hiddenInput).submit();
-
 			e.preventDefault();
+
+			/* Allow TemplateClassification extension to break submit */
+			if (win.enableTemplateClassificationEditorPlugin === true) {
+				require(['TemplateClassificationModalForce'], function forceTemplateClassificationModal(tcForce) {
+					if (tcForce.forceType() === true) {
```

Comment:
What if you reverse the condition? Then you wouldn't need the else {return;} part whatsoever :)